common/src/main/java/com/gregtechceu/gtceu/api/data/worldgen/generator/StandardVeinGenerator.java

```diff
@@ -150,15 +155,14 @@ public boolean generate(WorldGenLevel level, RandomSource random, GTOreDefinitio
                 if (y > level.getHeight(Heightmap.Types.OCEAN_FLOOR_WG, heightmapX, heightmapZ))
                     continue;
 
-                if (this.doPlaceNormal(level, random, entry, origin, this.blocks, minX, maxX, minZ, maxZ, minY, maxY, x, y, z, width, height)) {
-                    return true;
-                }
+                this.doPlaceNormal(generatedBlocks, random, entry, origin, this.blocks, minX, maxX, minZ, maxZ, minY, maxY, x, y, z, width, height);
```

Comment:
That's not possible anymore with the current code unfortunately, as the actual generation is deferred to when the individual chunks generate (and therefore can't check yet how many blocks were actually placed). Right here, only the `OreBlockPlacer`s for the individual block positions are created.

Is this a hard requirement for the ore vein shape or is it just for performance reasons?

I see a few possible options here:
- generate all placementsl as it is done here
- generate only the first placement that returns **possible** block positions to be generated later
- lock generation to the "variant" that actually produces blocks *first*, as soon as the vein's first chunk is generated.
  This would require additional (threadsafe) coordination though, so I'll only implement that if it's strictly necessary for vein shape
- limit size of `StandardVeinGenerator` to 3x3 chunks and use the old system for it (I'd rather not do that if possible)

Reply:
> is this a hard requirement for
both.

option 2 seems the best

Reply:
Then I'll implement the second one. Seems best to me as well, even though that brings a possibility of skipping ore veins if the blocks can't be generated later.